internal/transport/grpchttp2/framer.go

```diff
@@ -0,0 +1,311 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package grpchttp2 defines HTTP/2 types and a framer API and implementation.
+package grpchttp2
+
+import "golang.org/x/net/http2/hpack"
+
```

Comment:
Somewhere in this doc, it might be good to state that we plan to use these types only on the read path, and on the write path, we expect callers do pass in bytes that the framer will transform into HTTP/2 frames.